backend/tests/alert/test_alert.py

```diff
@@ -69,37 +71,281 @@ def override_delay(modules_names, before_func, before_kwargs):
+        self.assertEquals(push_notification, mock_push_notification.called)
+        self.assertTrue(r.notification_sent)
+        self.assertIsNotNone(r.notification_sent_at)
+        self.assertEqual("ADM", r.notification_sent_by)
+        self.assertTrue(r.close_notification_sent)
+        self.assertIsNotNone(r.close_notification_sent_at)
+        self.assertEqual("ADM", r.close_notification_sent_by)
+        self.assertFalse(r in get_active_registrations("CIS-160-001", TEST_SEMESTER))
+        self.assertFalse(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="O")
+        )
+        self.assertFalse(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="C")
+        )
+
+    def test_send_close_notification_push(self, mock_email, mock_text, mock_push_notification):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=True,
+            auto_resubscribe=True,
+            manual_resubscribe=False,
+        )
+
+    def test_send_close_notification_push_autoresub(
+        self, mock_email, mock_text, mock_push_notification
+    ):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=True,
+            auto_resubscribe=False,
+            manual_resubscribe=False,
+        )
+
+    def test_send_close_notification_autoresub(self, mock_email, mock_text, mock_push_notification):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=False,
+            auto_resubscribe=True,
+            manual_resubscribe=False,
+        )
+
+    def test_send_close_notification(self, mock_email, mock_text, mock_push_notification):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=False,
+            auto_resubscribe=False,
+            manual_resubscribe=False,
+        )
+
+    def test_send_close_notification_push_resub(
+        self, mock_email, mock_text, mock_push_notification
+    ):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=True,
+            auto_resubscribe=False,
+            manual_resubscribe=True,
+        )
+
+    def test_send_close_notification_resub(self, mock_email, mock_text, mock_push_notification):
+        self.send_close_notification_helper(
+            mock_email,
+            mock_text,
+            mock_push_notification,
+            push_notification=False,
+            auto_resubscribe=False,
+            manual_resubscribe=True,
+        )
+
+    def dont_resend_alert_helper(
```

Comment:
Ditto again. Will stop commenting but yeah, docstrings on helper functions would be amazing!